packages/react/src/components/FormattingToolbar/DefaultButtons/ImageCaptionButton.tsx

```diff
@@ -9,65 +14,60 @@ import {
 } from "react";
 import { RiText } from "react-icons/ri";
 
+import { useBlockNoteEditor } from "../../../editor/BlockNoteContext";
+import { useSelectedBlocks } from "../../../hooks/useSelectedBlocks";
 import { ToolbarButton } from "../../../components-shared/Toolbar/ToolbarButton";
 import { ToolbarInputDropdown } from "../../../components-shared/Toolbar/ToolbarInputDropdown";
 import { ToolbarInputDropdownButton } from "../../../components-shared/Toolbar/ToolbarInputDropdownButton";
 import { ToolbarInputDropdownItem } from "../../../components-shared/Toolbar/ToolbarInputDropdownItem";
-import { useSelectedBlocks } from "../../../hooks/useSelectedBlocks";
+import { checkBlockIsImage } from "./ReplaceImageButton";
 
-export const ImageCaptionButton = <BSchema extends BlockSchema>(props: {
-  editor: BlockNoteEditor<BSchema, any, any>;
-}) => {
-  const selectedBlocks = useSelectedBlocks(props.editor);
+export const ImageCaptionButton = () => {
+  const editor = useBlockNoteEditor<
+    BlockSchema,
+    InlineContentSchema,
+    StyleSchema
+  >();
 
-  const show = useMemo(
-    () =>
-      // Checks if only one block is selected.
-      selectedBlocks.length === 1 &&
-      // Checks if the selected block is an image.
-      selectedBlocks[0].type === "image" &&
-      // Checks if the block has a `caption` prop which can take any string
-      // value.
-      "caption" in props.editor.blockSchema["image"].propSchema &&
-      typeof props.editor.blockSchema["image"].propSchema.caption.default ===
-        "string" &&
-      props.editor.blockSchema["image"].propSchema.caption.values ===
-        undefined &&
-      // Checks if the block has a `url` prop which can take any string value.
-      "url" in props.editor.blockSchema["image"].propSchema &&
-      typeof props.editor.blockSchema["image"].propSchema.url.default ===
-        "string" &&
-      props.editor.blockSchema["image"].propSchema.url.values === undefined &&
-      // Checks if the `url` prop is not set to an empty string.
-      selectedBlocks[0].props.url !== "",
-    [props.editor.blockSchema, selectedBlocks]
-  );
+  const selectedBlocks = useSelectedBlocks(editor);
+
+  const imageBlock = useMemo(() => {
+    // Checks if only one block is selected.
+    if (selectedBlocks.length !== 1) {
+      return undefined;
+    }
+
+    const block = selectedBlocks[0];
+
+    if (checkBlockIsImage(block, editor)) {
+      return block;
+    }
+
+    return undefined;
+  }, [editor, selectedBlocks]);
 
   const [currentCaption, setCurrentCaption] = useState<string>(
-    show ? (selectedBlocks[0].props.caption as string) : ""
+    imageBlock ? imageBlock.props.caption : ""
   );
 
   useEffect(
-    () =>
-      setCurrentCaption(
-        show ? (selectedBlocks[0].props.caption as string) : ""
-      ),
-    [selectedBlocks, show]
+    () => setCurrentCaption(imageBlock ? imageBlock.props.caption : ""),
```

Comment:
I think this shouldn't run in an effect / maybe we don't need the `currentCaption` state?